### PR TITLE
Refactor metrics interface.

### PR DESCRIPTION
**Pull request by fejta**
Change Val to Values() and make it specific to logging metrics.
Create a reporter which only logs metric values periodically

### CODE REVIEW

File: util/metrics/log_test.go [hunk 1]
```diff
@@ -17,131 +17,226 @@ limitations under the License.
...
+				},
+			},
 		},
 		{
-			name:       "not enough fields",
-			n:          3,
-			fieldNames: []string{"user"},
-			fields:     []string{},
-			err:        true,
+			name:   "values",
```
Comment (michelle192837) [inline util/metrics/log_test.go:79]: nit: This is named the same as the last test case

**Review by michelle192837 — APPROVED** [review]
/hold

for comment

File: util/metrics/log_test.go (hunk 1 above)
Comment (fejta) [reply util/metrics/log_test.go]: Done

**Comment by fejta** [comment]
/hold cancel

Ready for another review

**Review by michelle192837 — APPROVED** [review]

**Pull request merged**